packages/widgets/src/widgets/TradeWidget/index.tsx

```
@@ -861,76 +855,127 @@ function TradeWidgetWrapped({
+        const proposedTargetToken = tokenList.find(
+          token => token.symbol.toLowerCase() === externalWidgetState.targetToken?.toLowerCase()
+        );
+
+        if (proposedTargetToken) {
+          // Check if current origin token would create an invalid pair
+          const isInvalidPair =
+            originToken &&
+            // Same token check
+            (originToken.symbol.toLowerCase() === proposedTargetToken.symbol.toLowerCase() ||
+              // Disallowed pair check (like ETH-WETH)
+              ((disallowedPairs || {})[originToken.symbol] &&
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)